defold-spine/src/comp_spine_model.cpp

```diff
@@ -467,19 +467,34 @@ namespace dmSpine
     {
         SpineModelComponent* component = (SpineModelComponent*)state->userData;
 
+        // Events are explained here: http://esotericsoftware.com/spine-api-reference#AnimationStateListener
         switch (type)
         {
-        // case SP_ANIMATION_START:
-        //     printf("Animation %s started on track %i\n", entry->animation->name, entry->trackIndex);
-        //     break;
-        // case SP_ANIMATION_INTERRUPT:
-        //     printf("Animation %s interrupted on track %i\n", entry->animation->name, entry->trackIndex);
-        //     break;
-        // case SP_ANIMATION_END:
-        //     printf("Animation %s ended on track %i\n", entry->animation->name, entry->trackIndex);
-        //     break;
+            // case SP_ANIMATION_START:
+            // {
+            //     printf("Animation %s started on track %i\n", entry->animation->name, entry->trackIndex);
+            //     break;
+            // }
+            // case SP_ANIMATION_INTERRUPT:
+            // {
+            //     printf("Animation %s interrupted on track %i\n", entry->animation->name, entry->trackIndex);
+            //     break;
+            // }
+            // case SP_ANIMATION_END:
+            // {
+            //     printf("Animation %s ended on track %i\n", entry->animation->name, entry->trackIndex);
+            //     break;
+            // }
             case SP_ANIMATION_COMPLETE:
             {
+                if (entry->mixingTo != 0)
+                {
+                    // While mixing, the previous animation completed.
+                    // The problem is that the track is already owned by the new animation, and we
+                    // cannot call the previous callback (as we haven't stored it)
+                    return;
+                }
```

Comment:
The issue was that the message (SP_ANIMATION_COMPLETE) comes _after_ we've already stored the information for the target animation.
Assuming we don't want to send an event from the previous animation (we have a new target animation, with a new callback), we can simply skip this event.

Reply:
It looks logical. It's more like a bugfix and I hope it will not be a breaking change for somebody